test/unit/specs/composables/use-window-scroll.spec.ts

```diff
@@ -0,0 +1,65 @@
+import Vue from 'vue'
+import { ref } from '@nuxtjs/composition-api'
+import { render } from '@testing-library/vue'
+
+import { useWindowScroll } from '~/composables/use-window-scroll'
+
+const getMockWindow = <T>(props: T) =>
+  ({
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    ...props,
+  } as unknown as typeof window)
+
+const UseWindowScrollTestContainer = Vue.component(
```

Comment:
Very interesting setup!

Reply:
I think it might be possible to create a "generic" component for this to test composables, but I'm not sure :sweat_smile: